matchingengine/OrderBook_test.go

```diff
@@ -139,41 +594,222 @@ func TestLoadOldBid(t *testing.T) {
+	models.LoadStocks()
+	time.Sleep(time.Second * 2)
 
-	mockBidQueue.EXPECT().Push(limitBid).Times(1)
-	mockDepth.EXPECT().AddOrder(false, false, stockPrice, stockQuantity).Times(1)
+	ob.processAsk(ask)
 
-	ob.LoadOldBid(limitBid)
+	if ob.asks.Head() != nil || ob.bids.Head() != nil {
+		l.Errorf("Errored in processAsk")
+	}
 
 }
 
-func TestLoadOldBidStopLoss(t *testing.T) {
+func Test_ProcessBid(t *testing.T) {
 
-	// This initializes the logger. This is required because LoadOldAsk tries accessing the logger
-	// and if it is not initialized, a nil reference error is thrown.
 	config := utils.GetConfiguration()
 	utils.Init(config)
+	models.Init(config, datastreams.GetManager())
 
-	mockControl := gomock.NewController(t)
-	defer mockControl.Finish()
+	var l = utils.Logger.WithFields(logrus.Fields{
+		"method": "Test_ProcessBid",
+	})
+
+	var stockID uint32 = 1
+	// var stockQuantity uint32 = 10
+	// var stockPrice uint32 = 20
+	var userID1 uint32 = 3
+	var userID2 uint32 = 4
+
+	var makeAsk = func(userId uint32, stockId uint32, isClosed bool) *models.Ask {
+		return &models.Ask{
+			UserId:   userId,
+			StockId:  stockId,
+			IsClosed: isClosed,
+		}
+	}
+
+	var makeBid = func(userId uint32, stockId uint32, isClosed bool) *models.Bid {
+		return &models.Bid{
+			UserId:   userId,
+			StockId:  stockId,
+			IsClosed: isClosed,
+		}
+	}
+
+	stock := &models.Stock{
+		Id: 1,
+	}
+
+	user1 := &models.User{Id: userID1, Cash: 2000000, Email: "test1@test.com"}
+	user2 := &models.User{Id: userID2, Cash: 2000000, Email: "test2@test.com"}
+
+	ask := makeAsk(3, 1, false)
+	bid := makeBid(4, 1, false)
+
+	db := utils.GetDB()
+
+	defer func() {
+		db.Exec("DELETE from OrderFills")
+		db.Exec("DELETE from Transactions")
+		db.Exec("DELETE from TransactionSummary")
+		db.Delete(ask)
+		db.Delete(bid)
+		db.Delete(user1)
+		db.Delete(user2)
+		db.Delete(stock)
+	}()
+
+	if err := db.Create(user1).Error; err != nil {
+		t.Fatal(err)
+	}
+
+	if err := db.Create(user2).Error; err != nil {
+		t.Fatal(err)
+	}
+
+	if err := db.Create(stock).Error; err != nil {
+		t.Fatal(err)
+	}
+
+	if err := db.Create(ask).Error; err != nil {
+		t.Fatal(err)
+	}
+
+	if err := db.Create(bid).Error; err != nil {
+		t.Fatal(err)
+	}
+
+	testAskQueue := NewAskPQueue(1)
+	testBidQueue := NewBidPQueue(0)
+	testAskStoplossQueue := NewAskPQueue(1)
+	testBidStoplossQueue := NewBidPQueue(0)
+	testDepth := datastreams.GetManager().GetMarketDepthStream(stockID)
+
+	ob := &orderBook{
+		logger: utils.Logger.WithFields(logrus.Fields{
+			"module":        "matchingengine.OrderBook.test",
+			"param_stockId": stockID,
+		}),
+		stockId:     stockID,
+		askChan:     make(chan *models.Ask),
+		bidChan:     make(chan *models.Bid),
+		asks:        testAskQueue,
+		bids:        testBidQueue,
+		askStoploss: testAskStoplossQueue,
+		bidStoploss: testBidStoplossQueue,
+		depth:       testDepth,
+	}
+
+	ob.LoadOldAsk(ask)
+	ob.LoadOldBid(bid)
+	models.LoadStocks()
+	time.Sleep(time.Second * 2)
```

Comment:
Use wg.SyncGroup as mentioned above

Reply:
time.Sleep was not necessary my bad. so just removing it works.